pkg/daemon/certificate_writer.go

```diff
@@ -110,6 +115,36 @@ func (dn *Daemon) syncControllerConfigHandler(key string) error {
 		pathToData[caBundleFilePath] = kubeAPIServerServingCABytes
 		pathToData[cloudCABundleFilePath] = cloudCA
 
+		cm, err := dn.clusterCmLister.ConfigMaps("openshift-config-managed").Get("kube-apiserver-server-ca")
+		if err != nil {
+			return err
+		}
+
+		data, err := cmToData(cm, "ca-bundle.crt")
+
+		if data != nil {
+			currentKC := clientcmdv1.Config{}
+			kcBytes, err := os.ReadFile(kubeConfigPath)
+			if err != nil {
+				return err
+			}
+			if kcBytes != nil {
```

Comment:
kcBytes is going to be guaranteed to be not nil, so this if statement is not needed.

Reply:
it seems  like it could be empty, e.g. if a runaway admin emptied the file.  But if we handle the empty case, we'd want to reset the entire file, with logic like [this][1].

[1]: https://github.com/openshift/machine-config-operator/blob/master/pkg/server/cluster_server.go#L153-L195

Reply:
we can be a little less careful here I guess, but i was running into some strange errors in the non-backported version so I added this in